### ISSUE TITLE
NamedTuple single line declaration gives error "Too many arguments for NamedTuple() [misc]

### ISSUE THREAD

**Issue by yesthesoup** [python/mypy#11047]
**Bug Report**

typing.NamedTuple one line declaration gives incorrect error

**To Reproduce**

```python
# test.py
from typing import NamedTuple

ExampleClass = NamedTuple('ExampleClass', test_id=str, timestamp=str)
```

```bash
mypy --show-error-codes test.py

test.py:[line-no]: error: Too many arguments for NamedTuple()  [misc]
```
**Expected Behavior**

mypy doesn't give this error

given the above syntax is valid python and works fine, I would expect it to not throw an error, like the following old structure declaration does not, which is labeled as "back-ward compatible usage" [in the official doc](https://docs.python.org/3/library/typing.html#typing.NamedTuple)

```python
ExampleClass = NamedTuple('ExampleClass', [(test_id, str), (timestamp, str)])
```

**Actual Behavior**

```bash
error: Too many arguments for NamedTuple()  [misc]
```

**Your Environment**

```
mypy --version
mypy 0.910

python --version
3.8.10
```

**Comment by sobolevn** [comment]
Working on it. It might take some time though.

**Comment by sobolevn** [comment]
There are multiple other bugs / missing features.

1. `fullname` must be `str` only, right now `bytes` is allowed by `mypy`, but not `cpython`:

```python
>>> import collections
>>> collections.namedtuple(b'custom', ['x'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sobolev/.pyenv/versions/3.9.1/lib/python3.9/collections/__init__.py", line 390, in namedtuple
    raise ValueError('Type names and field names must be valid '
ValueError: Type names and field names must be valid identifiers: "b'custom'"
```

**Comment by sobolevn** [comment]
Status update: I am rewritting argument parsing in `semanal_namedtuple` completely. It was way to complicated.

Right now I can parse all possible valid combinations of `NamedTuple` and its arguments.
Something like:

```python
def _parse_typing_namedtuple(self, typename: str) -> NamedTupleStructure:
        """Parsing ``typing.NamedTuple()`` call.

        In all of the examples below arguments can be named or positional.

        Possible valid cases:
        1. ``N = NamedTuple('N')`` - empty namedtuple,
        2. ``N = NamedTuple('N', [('a', int)])``
        3. ``N = NamedTuple(typename='N', fields=(('a', int),))``
        4. ``N = NamedTuple('N', a=int)`` with kwargs

        Corner cases, but still valid:
        7. ``N = NamedTuple('N', ())``, - empty namedtuple,
           we also count ``[]`` here
        6. ``N = NamedTuple('N', fields=None, a=int)``,
           in this case ``fields`` will not be present as a namedtuple field
        7. ``N = NamedTuple('N', None, a=int)``
        8. ``N = NamedTuple(a=int, typename='N')``
           kw-arguments can be in a different order, that's fine

        Everything else is considered **invalid**.
        We only accept statically known types.
        """
```
